### PR TITLE
Apply rate limiting to countries other than the target country

### PR DESCRIPTION
**Pull request by pyama86**
指定した国以外にレートリミットを書けるケースを想定しています。サービスを提供する国以外からのアクセスに対して厳しいレートリミットをかけたいことがあります。

I am considering cases where rate limiting is applied to countries other than the specified one. Sometimes, we want to impose strict rate limits on access from countries where our service is not provided.

### CODE REVIEW

**Review by k1LoW — APPROVED** [review]
LGTM!

Another way to implement this would be to provide an API similar to hosts.allow/hosts.deny or iptables.

It can be realized with two []strings and a wildcard (`*`).

### Example

```go
type CountryLimiter struct {
	db            *maxminddb.Reader
	countries     []string
	skipCountries []string
	BaseLimiter
}
```
Limit countries other than JP.

```go
CountryLimiter{
  db: db,
  countries: ["*"],
  skipCountries: ["JP"],
}
```

Pros of this proposal is that the meaning of the fields does not change.

It is also easy to understand if we want to change the settings.

Cons are that skipCountries is redundant.
This is because skipCountries is usually used only when expressing "other than ~".

File: country_limiter.go [hunk 1]
```diff
@@ -13,8 +13,9 @@ import (
 )
 
 type CountryLimiter struct {
-	db        *maxminddb.Reader
-	countries []string
+	db                         *maxminddb.Reader
+	limitRateForOtherCountries bool
```
Comment (k1LoW) [inline country_limiter.go]: Just an idea, how about specifying the type?

```suggestion
	CondType CondType
```

```go
type CondType bool
var Target CondType = true
var Excect CondType = false
```

File: country_limiter.go [hunk 2]
```diff
@@ -13,8 +14,9 @@ import (
 )
 
 type CountryLimiter struct {
-	db        *maxminddb.Reader
-	countries []string
+	db            *maxminddb.Reader
+	countries     map[string]bool
+	skipCountries map[string]bool
```
Comment (k1LoW) [inline country_limiter.go]: Using `struct{}` is memory efficient.

```suggestion
	countries     map[string]struct{}
	skipCountries map[string]struct{}
```

ref: https://speakerdeck.com/izzii/hong-tuteli-jie-suru-go-konpairazui-shi-hua-go-conference-mini-2023?slide=12

File: country_limiter.go [hunk 3]
```diff
@@ -58,14 +75,33 @@ func (l *CountryLimiter) Rule(r *http.Request) (*rl.Rule, error) {
...
+	}
+
+	if _, ok := l.countries[country]; ok {
+		return &rl.Rule{
+			Key:       remoteAddr,
+			ReqLimit:  l.reqLimit,
+			WindowLen: l.windowLen,
+		}, nil
 	}
 	return &rl.Rule{ReqLimit: -1}, nil
```
Comment (k1LoW) [inline country_limiter.go]:
```suggestion
	return noLimit, nil
```

File: country_limiter.go (hunk 2 above)
Comment (pyama86) [reply country_limiter.go]: 便利っぽい

**Review by k1LoW — APPROVED** [review]
GREAT! LGTM!!

**Pull request merged**